salt/_grains/caasp_nodename.py

```diff
@@ -2,15 +2,13 @@
 
 import salt.grains.core
 
-assert '__opts__' in globals()
-salt.grains.core.__opts__ = __opts__
-
-
 def caasp_nodename():
     """
     Override core.hostname grain to provide lowercased items
     localhost, fqdn, host, domain, nodename
     """
+    assert '__opts__' in globals()
+    salt.grains.core.__opts__ = __opts__
```

Comment:
I know this is intended more as a test than anything else, but I don't like the idea of setting `salt.grains.core.__opts__` here.

Given that `__opts__` should be available inside the grain function, and calling  `os_grains()` has proved to give us trouble regarding `__opts__`, what doesn't really match up to the idea claimed that `__opts__` should be available inside the function; I'd say the safest bet for us is to go straight the way salt goes: https://github.com/saltstack/salt/blob/3e844ed1df13fea1b8fbc4d5b9d5b4f130743364/salt/grains/core.py#L1187-L1188.

```
(_, grains['nodename'], _, version, _, _) = platform.uname()
```

and then `lower` that.